core/src/snapshot/postprocessers/postprocess.rs

```diff
@@ -0,0 +1,177 @@
+use heimdall_common::utils::strings::{
+    classify_token, find_balanced_encapsulator, tokenize, TokenType,
+};
+
+pub fn remove_double_negation(line: &str) -> String {
+    let mut cleaned = line.to_owned();
+
+    while cleaned.contains("((!((!((") {
```

Comment:
It would be more accurate to do some sort of tokenizer for the conditional, removing operations where the parentheses are unneeded first. Then, you can replace based additional `!`s with a regex replace.